impl/src/main/java/org/apache/myfaces/push/Json.java

```diff
@@ -111,12 +112,45 @@ else if (object instanceof Class<?>)
         {
             encode(((Class<?>) object).getName(), builder);
         }
+        else if (object instanceof Record)
+        {
+            encodeRecord(object, builder);
+        }
         else
         {
             encodeBean(object, builder);
         }
     }
 
+    private static void encodeRecord(Object recordObject, StringBuilder builder) {
+
+        builder.append('{');
+        boolean fieldsFound = false;
+
+        for (Field field : recordObject.getClass().getDeclaredFields())
+        {
+            field.setAccessible(true);
+            fieldsFound = true;
+            String name = field.getName();
+            builder.append(name);
+            builder.append(':');
+            try {
+                encode(field.get(recordObject), builder);
+            } catch (IllegalArgumentException e) {
+                e.printStackTrace();
+            } catch (IllegalAccessException e) {
+                e.printStackTrace();
+            }
```

Comment:
I'm not sure what I should do with error handling here.

Reply:
I feel like we should throw some sort of unchecked exception here right?